pkg/objstore/bos/bos_test.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package bos
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+func TestBOSUpload(t *testing.T) {
```

Comment:
I think it is already covered by e2e tests :) that's why you have the NewTestBucket() function. If you have updated `pkg/objstore/objtesting/foreach.go` then it should be possible to run the test suite via that. Does it not work for some reason?

Reply:
I'm not sure whether e2e tests can cover that case where the obj are very large, so i do it.

Reply:
Can we then add such test case to `AcceptanceTest` function in  `pkg/objstore/testing.go:82`? 🤗

Reply:
Thanks for your advice, I have added this case to AcceptanceTest.